app/mci/pkg/gcp/backendservice/backendservicesyncer.go

```diff
@@ -85,18 +89,43 @@ func (b *BackendServiceSyncer) ensureBackendService(lbName string, port ingressb
 		if backendServiceMatches(desiredBE, existingBE) {
 			// Nothing to do. Desired backend service exists already.
 			fmt.Println("Desired backend service exists already")
-			return nil
+			return existingBE, nil
 		}
 		// TODO (nikhiljindal): Require explicit permission from user before doing this.
 		fmt.Println("Updating existing backend service", name, "to match the desired state")
-		return b.bsp.UpdateGlobalBackendService(desiredBE)
+		return b.updateBackendService(desiredBE)
 	}
 	glog.V(5).Infof("Got error %s while trying to get existing backend service %s", err, name)
 	// TODO(nikhiljindal): Handle non NotFound errors. We should create only if the error is NotFound.
 	// Create the backend service.
 	fmt.Println("Creating backend service", name)
 	glog.V(5).Infof("Creating backend service %v", *desiredBE)
-	return b.bsp.CreateGlobalBackendService(desiredBE)
+	return b.createBackendService(desiredBE)
+}
+
+// updateBackendService updates the backend service and returns the updated backend service.
+func (h *BackendServiceSyncer) updateBackendService(desiredBE *compute.BackendService) (*compute.BackendService, error) {
+	name := desiredBE.Name
+	fmt.Println("Updating existing backend service", name, "to match the desired state")
+	err := h.bsp.UpdateGlobalBackendService(desiredBE)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("Backend service", name, "updated successfully")
+	return h.bsp.GetGlobalBackendService(name)
+}
+
+// createBackendService creates the backend service and returns the created backend service.
+func (h *BackendServiceSyncer) createBackendService(desiredBE *compute.BackendService) (*compute.BackendService, error) {
+	name := desiredBE.Name
+	fmt.Println("Creating backend service", name)
+	glog.V(5).Infof("Creating backend service %v", desiredBE)
+	err := h.bsp.CreateGlobalBackendService(desiredBE)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("Backend service", name, "created successfully")
```

Comment:
Unless you can guarantee that Get will never fail after a successful Create, it may be confusing to see "created successfully".

Reply:
Our cloud provider implementation waits for the operation to complete: https://github.com/kubernetes/kubernetes/blob/68d2722be072d229a347585fa7f964bf54b38c95/pkg/cloudprovider/providers/gce/gce_backendservice.go#L92.
So Get is guaranteed not to fail, unless of course you delete it first :)